service/security/shiro/src/main/java/org/eclipse/kapua/service/authorization/shiro/AuthorizationServiceImpl.java

```diff
@@ -29,6 +32,28 @@
 @KapuaProvider
 public class AuthorizationServiceImpl implements AuthorizationService {
 
+    @Override
+    public boolean[] isPermitted(List<Permission> permissions) throws KapuaException {
+        KapuaSession session = KapuaSecurityUtils.getSession();
+
+        if (session == null) {
+            throw new KapuaUnauthenticatedException();
+        }
+        if (session.isTrustedMode()) {
+            boolean[] returnedPermissions = new boolean[permissions.size()];
+            for (int i=0; i<permissions.size(); i++) {
+                returnedPermissions[i] = true;
+            }
+            return returnedPermissions;
+        }
+        else {
+            List<org.apache.shiro.authz.Permission> permissionsShiro = permissions.stream()
+                    .map(permission -> (org.apache.shiro.authz.Permission) permission)
+                    .collect(Collectors.toList());
+            return SecurityUtils.getSubject().isPermitted(permissionsShiro);
+        }
+    }
+
     @Override
     public boolean isPermitted(Permission permission)
```

Comment:
This should invoke `isPermitted(Lists.newArrayList(permission)` so we can use the same logic and we don't need to keep methods aligned

Reply:
Manipulating data to reuse code (also by just invoking an object instantiation), in a critical path, is not a good practice in my opinion (the performances will be affected)

Reply:
This will result in duplicated code. Next "sonar fixes branch" you know that this will be changed anyway....

Reply:
won't fix option can be used